src/main/java/com/pagsestagio/movieapi/service/UsuarioService.java

```diff
@@ -0,0 +1,108 @@
+import com.pagsestagio.movieapi.security.config.SecurityConfiguration;
+import com.pagsestagio.movieapi.security.userDetails.UserDetailsImpl;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UsuarioService {
+
+  private AuthenticationManager authenticationManager;
+
+  private JwtTokenService jwtTokenService;
+
+  private UsuarioRepository usuarioRepository;
+
+  private SecurityConfiguration securityConfiguration;
+
+  public UsuarioService(
+      AuthenticationManager authenticationManager,
+      JwtTokenService jwtTokenService,
+      UsuarioRepository usuarioRepository,
+      SecurityConfiguration securityConfiguration) {
+    this.authenticationManager = authenticationManager;
+    this.jwtTokenService = jwtTokenService;
+    this.usuarioRepository = usuarioRepository;
+    this.securityConfiguration = securityConfiguration;
+  }
+
+  public UsuarioRespostaRetornaTokenOuMensagem autenticarUsuario(LoginUsuarioDTO loginUsuarioDTO) {
+    UsuarioRespostaRetornaTokenOuMensagem retornoAutenticacaoUsuario = null;
+
+    Optional<Usuario> usuarioExistente =
+        usuarioRepository.findByNomeUsuario(loginUsuarioDTO.nomeUsuario());
+
+    if (loginUsuarioDTO.nomeUsuario() == null || loginUsuarioDTO.senha() == null) {
+      retornoAutenticacaoUsuario =
+          new UsuarioRespostaRetornaTokenOuMensagem(
+              null,
+              "Não foi possível fazer login com estas credenciais. Verifique os dados informados.");
+    } else if (usuarioExistente.isEmpty()) {
+      retornoAutenticacaoUsuario =
+          new UsuarioRespostaRetornaTokenOuMensagem(
+              null, "Usuário não cadastrado. Tente criar um usuário.");
+    } else {
+      UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken =
+          new UsernamePasswordAuthenticationToken(
+              loginUsuarioDTO.nomeUsuario(), loginUsuarioDTO.senha());
+      Authentication authentication =
+          authenticationManager.authenticate(usernamePasswordAuthenticationToken);
+      UserDetailsImpl userDetails = (UserDetailsImpl) authentication.getPrincipal();
+      String token = jwtTokenService.generateToken(userDetails);
+
+      retornoAutenticacaoUsuario = new UsuarioRespostaRetornaTokenOuMensagem(token, null);
+    }
+
+    return retornoAutenticacaoUsuario;
+  }
+
+  public UsuarioRespostaRetornaUsuarioOuMensagem criarUsuario(CriacaoUsuarioDTO criacaoUsuarioDTO) {
+    UsuarioRespostaRetornaUsuarioOuMensagem retornoCriacaoDeUsuario = null;
+
+    Optional<Usuario> usuarioExistente =
+        usuarioRepository.findByNomeUsuario(criacaoUsuarioDTO.nomeUsuario());
+
+    if (criacaoUsuarioDTO.nomeUsuario() == null
```

Comment:
ao invés de fazer isso aqui:

criacaoUsuarioDTO.nomeUsuario() == null
        || criacaoUsuarioDTO.senha() == null
        || criacaoUsuarioDTO.funcao() == null
        
daria para fazer algo como 
criacaoUsuarioDTO.isInvalid()


public boolean isInvalid() {
 return this.nomeUsuario == null || this.senha == null || this.funcao == null
}

https://medium.com/@functionoliveira/breve-resumo-sobre-os-princ%C3%ADpios-tell-dont-ask-e-fail-fast-358680f0db6c